src/main/java/com/pi4j/crowpi/components/UltrasonicDistanceSensorComponent.java

```diff
@@ -76,12 +150,85 @@ public double measure(double temperature) {
         return calculateDistance(pulseLength, temperature);
     }
 
+    /**
+     * Sets or disables the handler for the object found recognition.
+     * This event gets triggered whenever a object comes into the specified range
+     * Only a single event handler can be registered at once.
+     *
+     * @param min Minimum distance to the Object in cm
+     * @param max Maximum distance to the object in cm
+     * @param handler Event handler to call or null to disable
+     */
+    public void onObjectFound(double min, double max, SimpleEventHandler handler) {
+        this.onObjectFound(min, max, DEFAULT_TEMPERATURE, handler);
+    }
+
+    /**
+     * Sets or disables the handler for the object found recognition.
+     * This event gets triggered whenever a object comes into the specified range
+     * Only a single event handler can be registered at once.
+     *
+     * @param min Minimum distance to the Object in cm
+     * @param max Maximum distance to the object in cm
+     * @param temperature Temperature the sensor is operating at [°C]
+     * @param handler Event handler to call or null to disable
+     */
+    public void onObjectFound(double min, double max, double temperature, SimpleEventHandler handler) {
+        this.minRange = min;
+        this.maxRange = max;
+        this.temperature = temperature;
+        this.objectFoundHandler.set(handler);
+
+        if (handler != null) {
+            startPoller(DEFAULT_POLLER_PERIOD_MS);
+        } else {
+            stopPoller();
+        }
```

Comment:
I like the idea of automatically starting/stopping the poller on demand, as it avoids conflicts with manual polling done by the user and saves resources. The current implementation has a slight error though, as the following order of events will NOT work:

1. Register an event handler using `onObjectFound(0, 50, () -> System.out.println("Hi"));`
2. Register an event handler using `onObjectDisappeared(0, 50, () -> System.out.println("Hi"));`
3. Deregister the second event handler using `onObjectDisappeared(0, 0, null);`

The third operation will now stop the poller despite the first event handler still being active. I think this logic should be moved into a separate method which checks both possible event handler stores to decide if the poller should be executed or not.